### PR TITLE
Move start-up logic to the abstraction layer.

### PR DESCRIPTION
**Pull request by PiotrSikora**
Signed-off-by: Piotr Sikora <piotrsikora@google.com>

### CODE REVIEW

File: source/extensions/common/wasm/wasm.cc [hunk 1]
```diff
@@ -2240,12 +2243,21 @@ bool Wasm::initialize(const std::string& code, bool allow_precompiled) {
...
-  wasm_vm_->start(vm_context_.get());
+  startVm(vm_context_.get());
   code_ = code;
   allow_precompiled_ = allow_precompiled;
   return true;
 }
 
+void Wasm::startVm(Context* root_context) {
+  /* Call "_start" function, and fallback to "__wasm_call_ctors" if the former is not available. */
+  if (_start_) {
```
Comment (jplevyak) [inline source/extensions/common/wasm/wasm.cc]: We used to call both of these if there were available.  So _start calls __wasm_call_ctors ?

Comment (jplevyak) [reply source/extensions/common/wasm/wasm.cc]: See https://github.com/rustwasm/wasm-bindgen/issues/1216

Comment (PiotrSikora) [reply source/extensions/common/wasm/wasm.cc]: Yes, `_start` calls `__wasm_call_ctors`, so we don't want call it twice.

But we never called both (`_start` was added in `1.39.0`). We used to look for a "start" function, but none of the toolchains (Emscripten, WASI, Rust) use this WebAssembly indicator, and it's not supported by V8 anyway.

Comment (PiotrSikora) [reply source/extensions/common/wasm/wasm.cc]: We don't use wasm-bindgen (it's for JavaScript<>Rust bindings).

**Pull request merged**